imagetest/test_suites/guestagent/snapshot_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 Google LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     https://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build cit
+// +build cit
+
+package guestagent
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestSnapshotScripts(t *testing.T) {
+	utils.LinuxOnly(t)
+	ctx := utils.Context(t)
+	// Make snapshots directory and write pre and post snapshot scripts
+	err := os.MkdirAll("/etc/google/snapshots", 0770)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = os.WriteFile("/etc/google/snapshots/pre.sh", []byte("#!/bin/bash\ndate>>/etc/google/snapshots/pre-snapshot-write\n"), 0770)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = os.WriteFile("/etc/google/snapshots/post.sh", []byte("#!/bin/bash\ndate>>/etc/google/snapshots/post-snapshot-write\n"), 0770)
+	if err != nil {
+		t.Fatal(err)
+	}
+	prj, zone, err := utils.GetProjectZone(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	inst, err := utils.GetMetadata(ctx, "instance", "name")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Enable snapshot scripts in the agent and restart it.
+	// Wait 5 seconds for connection to snapshot service
+	agentcfg, err := os.ReadFile("/etc/default/instance_configs.cfg")
+	if err != nil {
+		t.Fatal(err)
+	}
+	agentcfg = append(agentcfg, []byte("[Snapshots]\nenabled = true\ntimeout_in_seconds = 300\n")...)
```

Comment:
why timeout of 300s?

Reply:
I set it to the maximum allowable to avoid flakes due to snapshot scripts taking longer than expected (not that I'm expecting this).

Worth nothing that 
1) test timeouts for CIT executions are orders of magnitude longer than this. 15 minutes would be a short timeout for CIT
2) we don't pass the test suite timeout down to the wrapper inside, so inside the VM we have no idea what the timeout is. The `utils.Context` wrapper actually always returns context.Background because of CIT semantics

Reply:
sounds good to me.